dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,368 @@
+                .withMimeType("text/plain")
+                .build();
+
+        // Fake up a request in REST form.
+        RequestItemRest rir = new RequestItemRest();
+        rir.setBitstreamId(bitstream.getID().toString());
+        rir.setItemId(item.getID().toString());
+        rir.setRequestEmail(RequestItemBuilder.REQ_EMAIL);
+        rir.setRequestMessage(RequestItemBuilder.REQ_MESSAGE);
+        rir.setRequestName(RequestItemBuilder.REQ_NAME);
+        rir.setAllfiles(false);
+
+        // Create it and see if it was created correctly.
+        ObjectMapper mapper = new ObjectMapper();
+        MvcResult mvcResult = getClient()
+                .perform(post(URI_ROOT)
+                        .content(mapper.writeValueAsBytes(rir))
+                        .contentType(contentType))
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$", Matchers.allOf(
+                        hasJsonPath("$.id", not(is(emptyOrNullString()))),
+                        hasJsonPath("$.type", is(RequestItemRest.NAME)),
+                        hasJsonPath("$.token", not(is(emptyOrNullString()))),
+                        hasJsonPath("$.requestEmail", is(RequestItemBuilder.REQ_EMAIL)),
+                        hasJsonPath("$.requestMessage", is(RequestItemBuilder.REQ_MESSAGE)),
+                        hasJsonPath("$.requestName", is(RequestItemBuilder.REQ_NAME)),
+                        hasJsonPath("$.allfiles", is(false)),
+                        hasJsonPath("$.requestDate", not(is(emptyOrNullString()))), // TODO should be an ISO datetime
+                        hasJsonPath("$._links.self.href", not(is(emptyOrNullString())))
+                )))
+                .andReturn();
+
+        // Clean up the created request.
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String,Object> map = mapper.readValue(content, Map.class);
+        String requestToken = String.valueOf(map.get("token"));
+        RequestItem ri = requestItemService.findByToken(context, requestToken);
+        requestItemService.delete(context, ri);
+
+        context.restoreAuthSystemState();
+    }
+
+    /**
+     * Test of getDomainClass method, of class RequestItemRepository.
+     */
+    @Test
+    public void testGetDomainClass() {
+        System.out.println("getDomainClass");
+        RequestItemRepository instance = new RequestItemRepository();
+        Class instanceClass = instance.getDomainClass();
+        assertEquals("Wrong domain class", RequestItemRest.class, instanceClass);
+    }
+
+    /**
+     * Verify that Spring Security's CSRF protection is working as we expect.
+     * We must test this using a simple non-GET request, as CSRF Tokens are not
+     * validated in a GET request.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testRefreshTokenWithInvalidCSRF()
```

Comment:
This method needs renaming & comments changed.  It's a test to verify that an item request cannot be made with an invalid CSRF token.  It's not about "RefreshToken"

Reply:
Oops, copy/paste error.  Done, but I don't see the issue with the comments.